### ISSUE TITLE
MvxObservableCollection cause Adapter to show wrong cells data.

### ISSUE THREAD

**Issue by Nickolas-** [MvvmCross/MvvmCross#2130]
Hi, found very interesting bug, when we are working with MvxObservableCollection, that help invoke events on UI thread it can cause problem with delayed event firing. 

So for example I receive concurrent 5 messages, I lock collection to insert some items. 4 messages will sit and wait until the first message inserted in right place. After I found place and insert a message from the background thread, it's very important, it invoke event , but it not ensure it will invoke synchronously because of this

```
protected override void OnCollectionChanged(NotifyCollectionChangedEventArgs e)
        {
            if (!EventsAreSuppressed)
            {
                InvokeOnMainThread(() => base.OnCollectionChanged(e));
            }
        }
```

Invoke on main thread under the hood uses a Application.SyhnronizationContext.Post(Action act).
In android specification you can find 

> post(Runnable r)
> Causes the Runnable r to be added to the message queue. 

So after my collection invoke event that the new message was added, the second message that was waiting get inserted too. And after small amount of time I get two notifications with

```
case System.Collections.Specialized.NotifyCollectionChangedAction.Add:
						NotifyItemRangeInserted(e.NewStartingIndex, e.NewItems.Count);
						break;
```

And it will invoke for creating two new cells, but the data will be wrong because , it will invoke two times with position for example 0. Then I will see that list containing two cells with same data.

Will later create sample project to test it. 

**Version:** 5.x

**Platform:** 
- [ ] Android

**Comment by nmilcoff** [comment]
I think this was solved in #2455. Can you please verify it @Nickolas-?

**Comment by Nickolas-** [comment]
Sorry for delayed response. Yes, try to repro with test project and it works correct at 6.2.0!
